app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt

```diff
@@ -92,6 +100,38 @@ class WebControlsTest {
         }
     }
 
+    @SmokeTest
+    @Test
+    fun emailLinkTest() {
+        webServer.enqueue(createMockResponseFromAsset("htmlControls.html"))
+        val htmlControlsPage = webServer.url("htmlControls.html").toString()
+
+        searchScreen {
+        }.loadPage(htmlControlsPage) {
+            progressBar.waitUntilGone(waitingTime)
+            clickLinkMatchingText("Email link")
+            progressBar.waitUntilGone(waitingTime)
+            clickOpenLinksInAppsOpenButton()
+            assertNativeAppOpens(GMAIL_APP, emailLink)
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun telephoneLinkTest() {
+        webServer.enqueue(createMockResponseFromAsset("htmlControls.html"))
+        val htmlControlsPage = webServer.url("htmlControls.html").toString()
+
+        searchScreen {
+        }.loadPage(htmlControlsPage) {
+            progressBar.waitUntilGone(waitingTime)
+            clickLinkMatchingText("Telephone link")
+            progressBar.waitUntilGone(waitingTime)
```

Comment:
Same here, on click, what are we waiting for here?

Reply:
Removed it, got misled by the fact that after tapping the re-direct link the progress bar is loading.